Kudu.Core/SiteExtensions/SiteExtensionManager.cs

```diff
@@ -50,6 +50,8 @@ public class SiteExtensionManager : ISiteExtensionManager
         private readonly IContinuousJobsManager _continuousJobManager;
         private readonly ITriggeredJobsManager _triggeredJobManager;
 
+        private static readonly string _toBeDeletedDirectoryPath = Path.Combine(System.Environment.ExpandEnvironmentVariables(@"%HOME%\data\SiteExtensions"), "tobedeleted");
```

Comment:
every time when we hit this issue, the end of the function will try to clean up "_toBeDeleted", so hopefully the next occurrence will cleanup the previous leftover data.

D:\local\Temp is a network drive, maybe i should correct my previous statement, seems like we can only move the lock file within the same drive. 
while working with Go we have the similar lock issue (later deployment try to update the exe from the first deployment which is lock by HttpPlatformHandler). it is easily repo to show you. and when try to  move the file over nextwork drive, it make a copy, and still keep the old one.  if we move within the same drive, it will actually be moved.